hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/KeyGenUtils.java

```diff
@@ -73,21 +73,16 @@ public static String getPartitionPathFromGenericRecord(GenericRecord genericReco
    */
   public static String[] extractRecordKeys(String recordKey) {
     String[] fieldKV = recordKey.split(",");
-    if (fieldKV.length == 1) {
-      return fieldKV;
-    } else {
-      // a complex key
-      return Arrays.stream(fieldKV).map(kv -> {
-        final String[] kvArray = kv.split(":");
-        if (kvArray[1].equals(NULL_RECORDKEY_PLACEHOLDER)) {
-          return null;
-        } else if (kvArray[1].equals(EMPTY_RECORDKEY_PLACEHOLDER)) {
-          return "";
-        } else {
-          return kvArray[1];
-        }
-      }).toArray(String[]::new);
-    }
+    return Arrays.stream(fieldKV).map(kv -> {
+      final String[] kvArray = kv.split(":");
+      if (kvArray[1].equals(NULL_RECORDKEY_PLACEHOLDER)) {
+        return null;
+      } else if (kvArray[1].equals(EMPTY_RECORDKEY_PLACEHOLDER)) {
+        return "";
+      } else {
+        return kvArray[1];
+      }
+    }).toArray(String[]::new);
```

Comment:
@wuwenchi could you add a unit test for the util method considering the fixed case?

Reply:
ok

Reply:
Why we generate a key like `name:val`, can we avoid that ?

Reply:
When a delete record is obtained, only `recordKey` is recorded in avro, and the fixed format of `recordKey` is `key:value`.

Reply:
Can you elaborate a little more for the details, a delete record still has a primary key right ?

Reply:
Here's what I know so far. If it is delete type data, hudi will use `DeleteRecord#create` to generate `DeleteRecord` according to the data. This class records the `HoodieKey` and then writes it to the avro log file. When the log is read, data of type `HoodieAvroRecord` is generated, and the `getRecordKey` method is called to obtain the primary key information of the data.

Reply:
Thanks, generally we should not use `Complex` key generators for single field primary key, but the fix makes the logic more robust,